src/pages/index.js

```diff
@@ -44,31 +84,113 @@ function openProfilePopup() {
 
 function changeProfileInfo(event, data) {
   user.setUserInfo(data);
-  editPopup.close();
+  api
+    .postUserInfo(data)
+    .then(() => {
+      editPopup.close();
+      toggleButtonText(editPopup.submitButton);
+    })
+    .catch((err) => console.log(err))
+    .finally(editPopup.resetSubmitText);
 }
 
 function addLocationCard(event, data) {
   const newLocation = { name: data["title"], link: data["image-url"] };
-  cardSection.renderer(newLocation);
-  event.target.reset();
-  addFormValidator.disableSubmitButton();
-  addPopup.close();
+  api
+    .postCard(newLocation)
+    .then((data) => {
+      cardSection.renderer(data);
+      event.target.reset();
+      addFormValidator.disableSubmitButton();
+      toggleButtonText(addPopup.submitButton);
+      addPopup.close();
+    })
+    .catch((err) => console.log(err))
+    .finally(() => {
+      addPopup.resetSubmitText();
+    });
 }
 
 function openImageModel(event) {
   imagePopup.open(event);
 }
 
-cardSection.renderItems(initialCards);
+function openEditProfilePicture() {
+  profilePicturePopup.open();
+}
+
+function openConfirmDelete(locationCardId, locationCard) {
+  deletePopup.cardId = locationCardId;
+  deletePopup.cardElement = locationCard;
+  deletePopup.open();
+}
+
+function deleteLocationCard(locationCardId, locationCard) {
+  api
+    .deleteCard(locationCardId)
+    .then(() => {
+      locationCard.remove();
+      deletePopup.cardId = "";
+      deletePopup.cardElement = "";
+      deletePopup.close();
+    })
+    .catch((err) => console.log(err));
+}
+
+function changeProfilePicture(event, pictureLink) {
+  user.setProfilePicture(pictureLink);
+  profilePicture.src = user.profilePicture.src;
```

Comment:
This is not needed

`UserInfo` should set the entire user data (including the **avatar**)